freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -115,6 +154,27 @@ def handle(self, event):
         bundles_nvrs = self._filter_bundles_by_pinned_related_images(
             auto_rebuild_nvrs)
 
+        # Now we have the bundle NVRs needs to be rebuilt, for each bundle NVR,
+        # construct the data of related images pullspec overrides
+        pullspec_overrides_by_nvr = {}
+        for bundle_nvr in bundles_nvrs:
+            images = [i for i in bundle_images if i['brew']['build'] == bundle_nvr]
```

Comment:
Rather than loop through `bundle_images` for each `nvr`, could you loop through it once outside of this for loop and create a dictionary of NVR to image. Then this for loop can be changed to iterate through that dictionary.